### PR TITLE
Patron: Updates the account upsell prompt

### PR DESCRIPTION
**Pull request by emilylaguna**
| 🛫 Depends on: https://github.com/Automattic/pocket-casts-ios/pull/851 | 🎨 [Designs](https://www.figma.com/file/PviUP0aiZctOprDuuYRwpb/Onboarding-%26-Upgrade-v2?type=design&node-id=3119-58186&t=2yK8L8dDKRfQtScR-4)
|:---:|:---:|

- Updates the design of the upsell view on the account view
- Adds an Upgrade Account item for Plus users
- Adds the ability to show the upgrade view showing a specific item by default
- Adds the ability to show a single item only on the upgrade view

## Screenshots

| Upgrade Prompt - Single Item | Upgrade Prompt - Multiple Items | Upgrade Account | Single Upgrade Item|
|:---:|:---:|:---:|:---:|
||||

### Upsell Single Item

1. In Xcode, enable the StoreKit configuration (Edit Scheme > Run > StoreKit Config)
2. Sign into an account without a subscription
3. Disable the patron feature flag if it's enable
4. Go to the Profile Tab
5. Tap on the Account button
6. ✅ Verify you see the new upsell with a single item

### Multiple Items

1. Enable the patron feature flag
2. Go back to the Account view
3. ✅ Verify you see 2 items in the upsell

### Showing the correct item

1. Tap on the Subscribe button for each of the items
2. When the upgrade view appears
3. ✅ Verify the default item displayed is the product you selected

### Upgrade Account

1. Subscribe to Plus
2. Go back to the account view
3. ✅ Verify you see the 'Upgrade Account' table item
4. Tap on it
5. ✅ Verify you see the upgrade view with Patron as the only item

### Purchasing
1. Subscribe to Patron
2. ✅ Verify the app reflects the new Patron state

## Checklist

- [ ] I have considered if this change warrants user-facing release notes and have added them to `CHANGELOG.md` if necessary.
- [ ] I have considered adding unit tests for my changes.
- [ ] I have updated (or requested that someone edit) [the spreadsheet](https://docs.google.com/spreadsheets/d/107jqrutZhU0fVZJ19SBqxxVKbV2NWSdQC9MFYdLiAxc/edit?usp=sharing) to reflect any new or changed analytics.

### CODE REVIEW

File: podcasts/AccountViewController+TableView.swift [hunk 1]
```diff
@@ -46,8 +49,24 @@ extension AccountViewController: UITableViewDataSource, UITableViewDelegate {
         case .upgradeView:
             let cell = tableView.dequeueReusableCell(withIdentifier: PlusAccountPromptTableCell.reuseIdentifier, for: indexPath) as! PlusAccountPromptTableCell
             cell.updateParent(self)
+            cell.contentSizeUpdated = { [weak self] size in
+                self?.upgradePromptViewSize = size
+            }
```
Comment (emilylaguna) [inline podcasts/AccountViewController+TableView.swift:54]: This is a bit hacky, but it works.

File: podcasts/AccountViewController+TableView.swift [hunk 2]
```diff
@@ -46,8 +49,24 @@ extension AccountViewController: UITableViewDataSource, UITableViewDelegate {
...
+
+            return cell
+
+        case .upgradeAccount:
+            let cell = tableView.dequeueReusableCell(withIdentifier: AccountViewController.actionCellId, for: indexPath) as! AccountActionCell
+            cell.cellLabel.text = "Upgrade Account"
```
Comment (charliescheer) [inline podcasts/AccountViewController+TableView.swift]: Should this be localized?

File: podcasts/Onboarding/Plus/Theme+Plus.swift [hunk 3]
```diff
@@ -1,11 +1,28 @@
...
+import UIKit
 
 extension Theme {
     var plusPrimaryColor: Color {
         Color(hex: "FEB525")
     }
 
     var patronPrimaryColor: Color {
-        Color(hex: "6046F5")
+        return Color(hex: ThemeConstants.patronHexLightTheme)
```
Comment (charliescheer) [inline podcasts/Onboarding/Plus/Theme+Plus.swift]: nitpick: I don't think you need the return here

**Review by charliescheer — APPROVED** [review]
Looks good to me

File: podcasts/Onboarding/Plus/Theme+Plus.swift (hunk 3 above)
Comment (emilylaguna) [reply podcasts/Onboarding/Plus/Theme+Plus.swift]: Oof, forgot to remove that. Fixed: d4875250c306cb92e65eafa43c7ebdd438c869ae

File: podcasts/AccountViewController+TableView.swift (hunk 2 above)
Comment (emilylaguna) [reply podcasts/AccountViewController+TableView.swift]: Yup! Fixed here: cd0e9a835ed0465c01dbadb8c26706410c5cf1e5

**Pull request merged**